server/appServices.js

```diff
@@ -138,6 +145,33 @@ async function watchKeyCloakSecrets(namespace, kubeclient) {
   secretStream.on('end', () => {
     watchKeyCloakSecrets(namespace, kubeclient);
   });
+
+  secretStream.on('close', () => {
+    retryWatch(namespace, kubeclient);
+  });
+}
+
+/**
+ * When the Kubernetes API is restarted, set a timeout and try to set up the watches after the API is up again.
+ *
+ * @param {String} namespace
+ * @param {*} kubeclient
+ */
+function retryWatch(namespace, kubeclient) {
+  console.log('connection to Kubernetes API lost');
+
+  if (connectAttempts < MAX_RETRIES) {
+    console.log(
+      `waiting ${RETRY_TIMEOUT} milliseconds before attempting services watches again`,
+      `attempt ${connectAttempts} of ${MAX_RETRIES}`
+    );
+    setTimeout(() => {
+      updateAppsAndWatch(namespace, kubeclient);
+    }, RETRY_TIMEOUT);
```

Comment:
this will only try the `updateAppsAndWatch` once, what happens if the api server is not back yet and the connection failed? Will it actually try again?

Reply:
Maybe something like [this](https://github.com/aerogear/mobile-developer-console/blob/688c7648ee09a49ae31f7b5a165d7155b9532dd9/src/services/crmanagers/PushVariantResourceManager.js#L5) would help here